File: phi_plugin/lib/Summary.py
import base64
import time

from .ByteReader import ByteReader


class Summary:
    """存档摘要"""

    def __init__(self, summary: str) -> None:
        """
        初始化

        :param summary: Base64编码的摘要数据
        """
        now = time.strftime("%Y %b.%d %H:%M:%S", time.localtime())
        self.updatedAt = now
        self.saveVersion = 0
        self.challengeModeRank = 0
        self.rankingScore = 0.0
        self.gameVersion = 0
        self.avatar = ""

        self.cleared: list[int] = [0] * 4
        self.fullCombo: list[int] = [0] * 4
        self.phi: list[int] = [0] * 4

        reader = ByteReader(base64.b64decode(summary))
        self.saveVersion = reader.getByte()
        self.challengeModeRank = reader.getShort()
        self.rankingScore = reader.getFloat()
        self.gameVersion = reader.getVarInt()
        self.avatar = reader.getString()
        for level in range(4):
            self.cleared[level] = reader.getShort()
            self.fullCombo[level] = reader.getShort()
            self.phi[level] = reader.getShort()

File: phi_plugin_koishi/lib/Summary.ts
// Assuming ByteReader is a class defined elsewhere
// Assuming Instant is a class imported from date-fns
// Assuming Base64 is an object imported from js-base64
import ByteReader from './ByteReader'
import Base64 from './Base64';


interface Summary {
    updatedAt: string;
    saveVersion: number;
    challengeModeRank: number;
    rankingScore: number;
    gameVersion: number;
    avatar: string;
    cleared: number[];
    fullCombo: number[];
    phi: number[];
}

class Summary {
    constructor(summary) {
        let now = Date().toString()
        let time = now.split(' ')
        this.updatedAt = `${time[3]} ${time[1]}.${time[2]} ${time[4]}`
        this.saveVersion = 0;
        this.challengeModeRank = 0;
        this.rankingScore = 0;
        this.gameVersion = 0;
        this.avatar = '';

        this.cleared = [];
        this.fullCombo = [];
        this.phi = [];

        const reader = new ByteReader(Base64.decode(summary));
        this.saveVersion = reader.getByte();
        this.challengeModeRank = reader.getShort();
        this.rankingScore = reader.getFloat();
        this.gameVersion = reader.getByte();
        this.avatar = reader.getString();
        this.cleared = [];
        this.fullCombo = [];
        this.phi = [];
        for (let level = 0; level < 4; level++) {
            this.cleared[level] = reader.getShort();
            this.fullCombo[level] = reader.getShort();
            this.phi[level] = reader.getShort();
        }
    }
}

export default Summary;